text_collector_examples/textfile_wrapper.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+#
+# Copyright 2018 The Prometheus Authors
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Description: Wrapper script to atomically write prometheus textfile script output.
+
+set -u
+
+usage() {
+  echo "usage: $(basename $0) <collector_script> <output_file>"
+  echo "NOTE: The output filename will automatically be written with a .prom extension"
```

Comment:
What's the rationale for that?

If we do this automagically, I'd at least check first whether the file already has a `.prom` suffix.

Reply:
Ehh, is it really an error to have `foo.prom.prom`?

Reply:
Or are you suggesting if not, add?

Reply:
It's not an error, but I'd intuitively specify the full path of the file and be surprised to see the double suffix.

I tend to be explicit and would error out if the file doesn't have a `.prom` suffix. I'd be fine with an implementation which adds it automatically if it's not there.

Reply:
I added an automatic append of `.prom` suffix.